ts/Core/Axis/OrdinalAxis.ts

```diff
@@ -720,15 +927,15 @@ namespace OrdinalAxis {
      * @param SeriesClass
      * Series class to use.
      */
-    export function compose(
-        AxisClass: typeof Axis,
-        ChartClass: typeof Chart,
-        SeriesClass: typeof Series
-    ): void {
+    export function compose<T extends typeof Axis>(
+        AxisClass: T,
+        SeriesClass: typeof Series,
+        ChartClass: typeof Chart
+    ): (typeof Composition&T) {
 
         AxisClass.keepProps.push('ordinal');
```

Comment:
The compose function needs double-compose protection. Create an non-exportet constant modifiedClasses, taking all class types in an array. Then do conditions around each of the three different class modifications to check whether the provided class argument is already in the array, otherwise modify and put the class in the array.